ssl/quic/quic_impl.c

```diff
@@ -384,6 +384,12 @@ SSL *ossl_quic_new(SSL_CTX *ctx)
         QUIC_RAISE_NON_NORMAL_ERROR(NULL, ERR_R_CRYPTO_LIB, NULL);
         goto err;
     }
+#if defined(OPENSSL_THREADS)
+    if ((qc->mutex = ossl_crypto_mutex_new()) == NULL) {
+        QUIC_RAISE_NON_NORMAL_ERROR(NULL, ERR_R_CRYPTO_LIB, NULL);
+        goto err;
+    }
+#endif
```

Comment:
Why is this necessary? Doesn't this get freed by `SSL_free(ssl_base)`?

Reply:
The mutex is needed to be there in case we call SSL_free(ssl_base) as it is being locked when it calls ossl_quic_free(). With the current ordering it can be not yet allocated when ossl_ssl_connection_new_int() below fails.